File: gobron-flutter-admin/lib/features/venue/models/venue.dart
import 'package:flutter/material.dart';

import '../../fields/models/field.dart';

class Venue {
  Venue({
    required this.id,
    required this.ownerId,
    required this.name,
    this.address,
    this.landmark,
    this.latitude,
    this.longitude,
    required this.openingTime,
    required this.closingTime,
    required this.workingDays,
    required this.isActive,
  });

  factory Venue.fromJson(Map<String, dynamic> json) {
    return Venue(
      id: json['id'] as int,
      ownerId: json['owner_id'] as int,
      name: json['name'] as String,
      address: json['address'] as String?,
      landmark: json['landmark'] as String?,
      latitude: (json['latitude'] as num?)?.toDouble(),
      longitude: (json['longitude'] as num?)?.toDouble(),
      openingTime: Field.timeFromJson(json['opening_time'] as String),
      closingTime: Field.timeFromJson(json['closing_time'] as String),
      workingDays: (json['working_days'] as List).cast<int>(),
      isActive: json['is_active'] as bool,
    );
  }

  final int id;
  final int ownerId;
  final String name;
  final String? address;
  final String? landmark;
  final double? latitude;
  final double? longitude;
  final TimeOfDay openingTime;
  final TimeOfDay closingTime;
  final List<int> workingDays;
  final bool isActive;
}

class VenueInput {
  VenueInput({
    required this.name,
    this.address,
    this.landmark,
    this.latitude,
    this.longitude,
    required this.openingTime,
    required this.closingTime,
    required this.workingDays,
    required this.isActive,
  });

  Map<String, dynamic> toJson() => {
    'name': name,
    'address': address,
    'landmark': landmark,
    'latitude': latitude,
    'longitude': longitude,
    'opening_time': Field.timeToJson(openingTime),
    'closing_time': Field.timeToJson(closingTime),
    'working_days': workingDays,
    'is_active': isActive,
  };

  final String name;
  final String? address;
  final String? landmark;
  final double? latitude;
  final double? longitude;
  final TimeOfDay openingTime;
  final TimeOfDay closingTime;
  final List<int> workingDays;
  final bool isActive;
}

File: gobron-backend/app/models/venue.py
"""Venue model — one physical sports complex managed by a field owner."""
from datetime import datetime, time

from sqlalchemy import (
    ARRAY,
    Boolean,
    DateTime,
    ForeignKey,
    Integer,
    String,
    Time,
    func,
)
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.database import Base


class Venue(Base):
    __tablename__ = "venues"

    id: Mapped[int] = mapped_column(primary_key=True)
    owner_id: Mapped[int] = mapped_column(
        ForeignKey("users.id", ondelete="CASCADE"), unique=True, index=True
    )

    name: Mapped[str] = mapped_column(String(150), nullable=False)
    address: Mapped[str | None] = mapped_column(String(255))
    landmark: Mapped[str | None] = mapped_column(String(255))
    latitude: Mapped[float | None] = mapped_column()
    longitude: Mapped[float | None] = mapped_column()

    opening_time: Mapped[time] = mapped_column(Time, nullable=False, default=time(8, 0))
    closing_time: Mapped[time] = mapped_column(Time, nullable=False, default=time(23, 0))
    working_days: Mapped[list[int]] = mapped_column(
        ARRAY(Integer), default=lambda: [0, 1, 2, 3, 4, 5, 6]
    )
    is_active: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)

    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now()
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), onupdate=func.now()
    )

    owner: Mapped["User"] = relationship()  # noqa: F821
    fields: Mapped[list["Field"]] = relationship(back_populates="venue")  # noqa: F821
